Helpers task290 improve on boarding docs

sonniki opened the pull request:
#290 
- Copied the missing `docs/onboarding/kaizenflow.set_up_development_environment.how_to_guide.md` from the kaizenflow repo
- Removed outdated references to ZenHub and ZH Epics and Pipelines
- Removed outdated references to Telegram
- Added `docs/onboarding/all.must_read_checklist.md`: a short-ish list of must-read docs that should be presented to anyone joining, be it with the full or light onboarding
    - TODO: Make sure all the docs in the list are onboarding-agnostic, i.e. they do not contain specific references to either the full onboarding process (e.g. developing on the server) or the light onboarding process (e.g. working as intern on Kaizenflow)
- Added a link to the must-read doc to `docs/onboarding/all.onboarding_checklist.md`
    - TODO: Create a "light onboarding checklist" doc with dev setup instructions for working on the laptop and the link to the same must-read list

gpsaggese commented on docs/coding/all.code_design.how_to_guide.md:
```diff
@@ -31,7 +31,7 @@
     - What should the code do?
     - What are the functionalities you want to implement?
-    - What are the functionalities you don’t want to implement? (what are you
+    - What are the functionalities you don't want to implement? (what are you
```
We could / should add a linter phase for some of these things. They often come from translating gdocs to markdown

sonniki replied:
This change is a Linter change already (not sure which phase, probably some 3rd-party lib wrapper)

gpsaggese commented on docs/onboarding/all.must_read_checklist.md:
```diff
@@ -0,0 +1,48 @@
+<!-- toc -->
```
all.must_read_checklist.reference.md 

(according to https://diataxis.fr/)

Also this is specific of developer (since it doesn't cover setting up and so on).
So maybe all.dev_must_read...?

Your call

gpsaggese commented on docs/onboarding/all.must_read_checklist.md:
```diff
@@ -0,0 +1,48 @@
+<!-- toc -->
+
+- [Must-read documents for a new team member](#must-read-documents-for-a-new-team-member)
```
Good stuff!

gpsaggese commented on docs/onboarding/all.onboarding_checklist.md:
```diff
@@ -139,7 +139,7 @@
 - [ ] **Team member**:
```
This is probably a reference, but whatever...

gpsaggese commented on docs/onboarding/kaizenflow.set_up_development_environment.how_to_guide.md:
```diff
@@ -0,0 +1,356 @@
+<!-- toc -->
```
maybe it's `all.set_up...` or it's for the "light on-boarding", but it's useful also for the full-time people.

Not sure how to solve it without refactoring. Thoughts?

sonniki replied:
@gpsaggese do you mind if I take a closer look at this file in another PR? As well as create the light onboarding checklist doc.
I got a bit carried away and this PR is now huge... Could you please take another look and merge if it looks okay to you?

sonniki replied:
Nevermind, did it in this PR anyway

gpsaggese commented on docs/work_organization/ck.github_projects_process.reference.md:
```diff
@@ -238,21 +233,14 @@ Next steps
-_Note: this is currently optional part, but can aid in a more comprehensive
-project management, it’s up to each project owner if they want to use this
-feature_
+  [`sprint_iteration.yml`](/github/workflows/sprint_iteration.yml) is set up to
```
missing .github

We should add a linter step to check that files exist. Unless we already have it. Can you pls file a bug, if needed?

sonniki replied:
We do have it. Linting this file produces
```
docs/work_organization/ck.github_projects_process.reference.md:235: '/github/workflows/sprint_iteration.yml' does not exist [fix_md_links]
```
I've missed it the first time around

sonniki replied:
Huh, turns out:
1) This file only exists in `cmamp` and doesn't exist in the same dir in `helpers`. This is a problem for Linter because even if the link is a full http link to `cmamp`, this Linter phase cuts off the "https://github.com/causify-ai/cmamp/blob/master" part and what is left automatically points to the local repo (`helpers`). To bypass it, I guess, we need to update the Linter phase so that it doesn't cut off the http part all the time but only if the repo name in the link is the same as the local repo. I'll file a bug.
2) Another thing is, apparently, it's the same Linter phase that removes the `.` from `.github` here. I'll file another bug =/

sonniki replied:
Filed 1) #300 and 2) #301

gpsaggese reviewed (commented):
Great work! Let's keep the docs in shape

sonniki commented:
@gpsaggese, UPD
- Added [`docs/onboarding/all.light_onboarding_checklist.reference.md`](https://github.com/causify-ai/helpers/blob/14e975f7e057dfccd34041f447e9be6379a1a287/docs/onboarding/all.light_onboarding_checklist.reference.md)
- Factored out the "technical" dev env set up part from that KaizenFlow doc into [`docs/onboarding/all.set_up_development_on_laptop.how_to_guide.md`](https://github.com/causify-ai/helpers/blob/14e975f7e057dfccd34041f447e9be6379a1a287/docs/onboarding/all.set_up_development_on_laptop.how_to_guide.md). This doc is now referenced in the light onboarding checklist.
- Renamed that KaizenFlow doc after moving the technical sections out -> [`docs/onboarding/kaizenflow.prepare_for_development.how_to_guide.md`](https://github.com/causify-ai/helpers/blob/14e975f7e057dfccd34041f447e9be6379a1a287/docs/onboarding/kaizenflow.prepare_for_development.how_to_guide.md)

gpsaggese reviewed (commented):
All good.

FYI One trick to avoid the monster PRs, when you realize that there is a problem, is to do a PR only running the linter and merge it. Then you pick up those changes from master and the PR has only your changes.
